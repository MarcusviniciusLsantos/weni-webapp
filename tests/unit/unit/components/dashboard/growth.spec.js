import { shallowMount, createLocalVue } from '@vue/test-utils';
import growth from '@/components/dashboard/growth.vue';
import i18n from '@/utils/plugins/i18n';

const localVue = createLocalVue();

describe('growth.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(growth, {
      localVue,
      i18n,
      mocks: {
        $t: () => 'some specific text',
      },
      stubs: {
        UnnnicCard: true,
      },
    });
  });

  it('renders a snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should return percentage', async () => {
    const res = await wrapper.vm.getPercentage(20, 10); // 10 to 20 -> 100%
    expect(res).toEqual(100);
  });

  it('should return null percentage', async () => {
    const res = await wrapper.vm.getPercentage(10, 0); // 10 to 20 -> 100%
    expect(res).toEqual(null);
  });
});
