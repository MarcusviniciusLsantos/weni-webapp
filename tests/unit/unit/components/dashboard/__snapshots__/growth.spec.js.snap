// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`growth.vue > renders a snapshot 1`] = `
<div
  class="weni-growth"
>
  
  <unnnic-card-stub
    icon="messages-bubble-1"
    modelvalue="75167071"
    percent="34097"
    scheme="aux-blue"
    title="Sent messages"
    type="dash"
  />
  <unnnic-card-stub
    icon="account_tree"
    modelvalue="16587"
    percent="861"
    scheme="aux-pink"
    title="Flows created"
    type="dash"
  />
  <unnnic-card-stub
    icon="chat-translate-1"
    modelvalue="184"
    percent="0"
    scheme="aux-purple"
    title="Languages supported"
    type="dash"
  />
  <unnnic-card-stub
    icon="chat-translate-1"
    modelvalue="5461975"
    percent="46223"
    scheme="aux-lemon"
    title="People reached"
    type="dash"
  />
  
</div>
`;
